config/config_test.go

```diff
@@ -232,3 +233,51 @@ func TestMinTLSVersion(t *testing.T) {
 		}
 	})
 }
+
+type testFormatter struct {
+	messages map[string]log.Level
+}
+
+func (tf *testFormatter) Format(entry *log.Entry) ([]byte, error) {
+	tf.messages[entry.Message] = entry.Level
+	return nil, nil
+}
+
+func TestDeprecatedFlags(t *testing.T) {
+	a := os.Args
+	o := log.StandardLogger().Out
+	f := log.StandardLogger().Formatter
+	defer func() {
+		os.Args = a
+		log.SetOutput(o)
+		log.SetFormatter(f)
+	}()
+
+	formatter := &testFormatter{messages: make(map[string]log.Level)}
+
+	os.Args = []string{
+		"skipper",
+		"-config-file=testdata/deprecated.yaml",
+		"-enable-prometheus-metrics=false", // set to default value and does not produce deprecation warning
```

Comment:
It would be better if logic detects use of default values as well - i.e. any use of deprecated options